modules/sfp_countryname.py

```diff
@@ -0,0 +1,323 @@
+            # Get country code from IBAN 
+            countryCode = srcIBAN[0:2].upper()
+            return tldCountryCodes[countryCode]
+        except:
+            # No country name is found in the IBAN
+            return None
+    
+    # Detect name of country from Who Is lookup data
+    def detectCountryFromWhoIs(self, srcWhoIs):
+        
+        # Get dictionary of country codes and  country names
+        abbvCountryCodes = self.getCountryCodeDict()
+        countries = set()
+
+        # Look for countrycodes and country in whois data
+        for countryName in abbvCountryCodes.values(): 
+
+            # Look for country name in whois data
+            matchCountries = re.findall("[\s,'\"]" + countryName + "[\s,'\"]", srcWhoIs, re.IGNORECASE)
+
+            if len(matchCountries) > 0:
+                # Get country name from first index of list
+                # Extract only the text part of the country code
+                matchCountry = matchCountries[0].strip(",").strip("'").strip("\"").strip()
+                countries.add(matchCountry)
+        
+        # If any countries are found
+        if len(countries) > 0:
+            return countries
+
+        return None
+            
+    # What events is this module interested in for input
+    def watchedEvents(self):
+        return ["IBAN_NUMBER", "PHONE_NUMBER", "AFFILIATE_DOMAIN_NAME",
+                "CO_HOSTED_SITE_DOMAIN", "DOMAIN_NAME", "SIMILARDOMAIN",
+                "AFFILIATE_DOMAIN_WHOIS", "CO_HOSTED_SITE_DOMAIN_WHOIS",
+                'DOMAIN_WHOIS']
+
+    # What events this module produces
+    def producedEvents(self):
+        return ["COUNTRY_NAME"]
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+        
+        evttype = "COUNTRY_NAME"
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        myres = list()
+        countryName = None
+
+        # Process the event data based on incoming event type
+        if eventName == "PHONE_NUMBER":
+            countryName = self.detectCountryFromPhone(eventData)
+        elif eventName in ["DOMAIN_NAME", "SIMILARDOMAIN"] or (eventName == "AFFILIATE_DOMAIN_NAME" and self.opts["affiliate"]) or(eventName == "CO_HOSTED_SITE_DOMAIN" and self.opts["coHosted"]):
+            countryName = self.detectCountryFromTLD(eventData)
+        elif eventName == "IBAN_NUMBER":
+            countryName = self.detectCountryFromIBAN(eventData)
+        elif eventName == "DOMAIN_WHOIS" or (eventName == "AFFILIATE_DOMAIN_WHOIS" and self.opts["affiliate"]) or (eventName == "CO_HOSTED_SITE_DOMAIN_WHOIS" and self.opts["coHosted"]):
+            countryNames = self.detectCountryFromWhoIs(eventData)
+
+        # If whois module data source is recevied, multiple country names might be there
+        if 'countryNames' in locals() and "WHOIS" in eventName:
```

Comment:
Not sure why you have `countryName` and `countryNames`.. seems like there's an opportunity to simplify there? Maybe just have `countryNames` which is a `list()` and for the functions returning single values, encapsulate the result in `[]`?